### PR TITLE
Enable vxsort for native AOT

### LINKED ISSUE #77: Reenable vxsort in GC
This was disabled due to missing symbols.

The VXSort feature enables intrinsic-optimized sorting within the garbage collector.

### PR DESCRIPTION
**Pull request by MartinZikmund**
Closes #77.

Hopefully, I changed the one-liner right 🙂.

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
```
Runtime.lib(gcwks.obj) : error LNK2019: unresolved external symbol "void __cdecl InitSupportedInstructionSet(int)" (?InitSupportedInstructionSet@@YAXH@Z) referenced in function "public: static long __cdecl WKS::gc_heap::initialize_gc(unsigned __int64,unsigned __int64,unsigned __int64)" (?initialize_gc@gc_heap@WKS@@SAJ_K00@Z) [F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.x64.Release\nativeaot\SmokeTests\Exceptions\Exceptions\nativebuild.proj]
Runtime.lib(gcwks.obj) : error LNK2019: unresolved external symbol "bool __cdecl IsSupportedInstructionSet(enum InstructionSet)" (?IsSupportedInstructionSet@@YA_NW4InstructionSet@@@Z) referenced in function "protected: static void __cdecl WKS::gc_heap::grow_mark_list(void)" (?grow_mark_list@gc_heap@WKS@@KAXXZ) [F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.x64.Release\nativeaot\SmokeTests\Exceptions\Exceptions\nativebuild.proj]
Runtime.lib(gcwks.obj) : error LNK2019: unresolved external symbol "void __cdecl do_vxsort_avx2(unsigned char * *,unsigned char * *,unsigned char *,unsigned char *)" (?do_vxsort_avx2@@YAXPEAPEAE0PEAE1@Z) referenced in function "protected: static void __cdecl WKS::gc_heap::plan_phase(int)" (?plan_phase@gc_heap@WKS@@KAXH@Z) [F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.x64.Release\nativeaot\SmokeTests\Exceptions\Exceptions\nativebuild.proj]
Runtime.lib(gcwks.obj) : error LNK2019: unresolved external symbol "void __cdecl do_vxsort_avx512(unsigned char * *,unsigned char * *,unsigned char *,unsigned char *)" (?do_vxsort_avx512@@YAXPEAPEAE0PEAE1@Z) referenced in function "protected: static void __cdecl WKS::gc_heap::plan_phase(int)" (?plan_phase@gc_heap@WKS@@KAXH@Z) [F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.x64.Release\nativeaot\SmokeTests\Exceptions\Exceptions\nativebuild.proj]
```

Looks like one more thing is needed.

Could you add the VXSORT sources to the CMakeLists of the runtime? Take the file list (including the "Windows only" and "AMD64 only" conditions) from here:

https://github.com/dotnet/runtimelab/blob/021f8f1a1d14ff4ac9c9fa16ecd9bbc18aafc56e/src/coreclr/src/gc/CMakeLists.txt#L42-L54

And put them somewhere here:

https://github.com/dotnet/runtimelab/blob/feature/NativeAOT/src/coreclr/src/nativeaot/Runtime/CMakeLists.txt

We don't have a GC_SOURCES in that CMake file, so maybe just append the file list to COMMON_RUNTIME_SOURCES (you should be able to see a couple sample entries when you look for `${GC_DIR}`).

Diky!

File: src/coreclr/src/nativeaot/Runtime/CMakeLists.txt [hunk 1]
```diff
@@ -193,6 +193,20 @@ else()
   )
 endif()
 
+if (CLR_CMAKE_TARGET_ARCH_AMD64 AND CLR_CMAKE_TARGET_WIN32)
+  set (APPEND COMMON_RUNTIME_SOURCES
```
Comment (MichalStrehovsky) [inline src/coreclr/src/nativeaot/Runtime/CMakeLists.txt:197]:
```suggestion
  list(APPEND COMMON_RUNTIME_SOURCES
```

Comment (MartinZikmund) [reply src/coreclr/src/nativeaot/Runtime/CMakeLists.txt]: oh! Thanks :-)

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**